src/main/java/org/opensearch/knn/index/codec/util/KNNCodecUtil.java

```diff
@@ -112,7 +113,17 @@ public static String buildEngineFilePrefix(String segmentName) {
     }
 
     public static String buildEngineFileSuffix(String fieldName, String extension) {
-        return String.format("_%s%s", fieldName, extension);
+        return String.format("_%s%s", escapeInvalidFileNameCharsInFieldName(fieldName), extension);
+    }
+
+    private static String escapeInvalidFileNameCharsInFieldName(final String fieldName) {
```

Comment:
if we are escaping the characters here, do we need to make changes here too: https://github.com/opensearch-project/k-NN/blob/27f31684e4672f5bdb7026ac06601c4a7a123743/src/main/java/org/opensearch/knn/index/query/KNNWeight.java#L348

when we are creating the path for this file for query?

Also, what will be the exact file name on the disk will it has the escape characters?

Reply:
Q1. `if we are escaping the characters here, do we need to make changes here too?`
Ans : Yes, I think so. Will include the change in weight class in this PR as well.


Q2. `what will be the exact file name on the disk will it has the escape characters?`
The file format would be `<Version>_<Escaped Field Name><Ext>`.

Ex:
Field name=`my_vector`, then file name `_0_2011_my_vector.hnswc`
Field name=`my vector` (it includes invalid char for filename), then the resulting file name would be `_0_2011_my.vector.hnswc`

Reply:
> Q1. if we are escaping the characters here, do we need to make changes here too?
Ans : Yes, I think so. Will include the change in weight class in this PR as well.

I wonder why this is not caught through our existing test? Could you add test if the case is missing?

Reply:
reason for that it, what being added is unit test and not IT. Hence this was missed.

Reply:
@0ctopus13prime lets also add the unit test for KNNWeight apart from IT where we are doing unit test for getEngineFiles(). This will ensure that nothing is missed

Reply:
Coming right up